support-frontend/app/controllers/AnalyticsController.scala

```diff
@@ -0,0 +1,71 @@
+package controllers
+
+import actions.CustomActionBuilders
+import actions.AsyncAuthenticatedBuilder.OptionalAuthRequest
+import com.gu.identity.model.User
+import com.gu.monitoring.SafeLogging
+import io.circe.{Encoder, Json}
+import io.circe.generic.semiauto.deriveEncoder
+import io.circe.syntax._
+import play.api.libs.circe.Circe
+import play.api.mvc._
+import play.api.mvc.Security.AuthenticatedRequest
+import services.{MParticleClient, MParticleUserProfile}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+case class AnalyticsUserProfileResponse(
+    identityId: String,
+    hasMobileAppDownloaded: Boolean,
+    hasFeastMobileAppDownloaded: Boolean,
+)
+
+object AnalyticsUserProfileResponse {
+  implicit val encoder: Encoder[AnalyticsUserProfileResponse] = deriveEncoder
+}
+
+class AnalyticsController(
+    components: ControllerComponents,
+    actionRefiners: CustomActionBuilders,
+    mparticleClient: MParticleClient,
+)(implicit val exec: ExecutionContext)
+    extends AbstractController(components)
+    with Circe
+    with SafeLogging {
+
+  import actionRefiners._
+  import AnalyticsUserProfileResponse._
+
+  type AuthenticatedUserRequest[A] = AuthenticatedRequest[A, User]
+
+  private val RequireAuthenticatedUser = new ActionRefiner[OptionalAuthRequest, AuthenticatedUserRequest] {
+    override protected def executionContext: ExecutionContext = exec
+
+    override protected def refine[A](
+        request: OptionalAuthRequest[A],
+    ): Future[Either[Result, AuthenticatedUserRequest[A]]] =
+      request.user match {
+        case Some(user) => Future.successful(Right(new AuthenticatedRequest(user, request)))
+        case None =>
+          logger.warn("getAnalyticsUserProfile called but user is not authenticated")
+          Future.successful(Left(Unauthorized("User must be authenticated to access this endpoint")))
+      }
+  }
```

Comment:
didn't realise there wasn't one already that's by other things! There used to be but I can't find one now.
 but this is a neat way to extract it 👍